src/main/java/com/bizo/awsstubs/services/dynamodb2/AttributeValueComparator.java

```diff
@@ -0,0 +1,259 @@
+package com.bizo.awsstubs.services.dynamodb2;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+
+public class AttributeValueComparator {
+
+  private AttributeValueComparator() {
+  }
+
+  // attribute1 == attribute2
+  public static boolean equal(AttributeValue attribute1, AttributeValue attribute2) {
```

Comment:
You could try something like:

```
interface AttributeValueComparator {
  boolean matchString(String a, String b);
  boolean matchByteBuffer(ByteBuffer a, ByteBuffer b);
  boolean matchBoolean(Boolean a, Boolean b);
  ...for M/N/NS/etc.
}

public class EqualComparator {
  @Override boolean matchString(a, b) { return a.equals(b); }
  @Override boolean matchByteBuffer(a, b) { return a.equals(b); }
  ...
}

public class LessThanComparator {
  @Override boolean matchString(a, b) { return a.compareTo(b) < 0; }
  ...
}

public static boolean equal(AV a1, AV a2) {
  return matches(new EqualComparator(), a1, a2);
}

public static boolean lessThan(AV a1, AV a2) {
  return matches(new LessThanComparator(), a1, a2);
}

...

public static boolean matches(AttributeValueComparator c, AV a1, AV a2) {
  return (a1 != null && a2 != null)
    && ((a1.getS() != null && a2.getS() != null && c.matchString(a1.getS(), a2.getS())
      || (a1.getB() != null && a2.getB() != null && c.matchByteBuffer(a1.getB(), a2.getB())
      || ...);
}
```

That way you have only one ugly "if S, use a1.getS/a2.getS, if B, use a1.getB, a2.getB, etc."